cedar-policy/src/api.rs

```diff
@@ -4496,14 +4599,321 @@ mod ancestors_tests {
+                ("numDirectReports".into(), RestrictedExpression::new_long(3)),
+                (
+                    "department".into(),
+                    RestrictedExpression::new_string("Sales".into()),
+                ),
+                (
+                    "manager".into(),
+                    RestrictedExpression::from_str(r#"Employee::"456""#).unwrap(),
+                ),
+                ("hr_contacts".into(), RestrictedExpression::new_set([])),
+                (
+                    "json_blob".into(),
+                    RestrictedExpression::new_record([
+                        ("inner1".into(), RestrictedExpression::new_bool(false)),
+                        (
+                            "inner2".into(),
+                            RestrictedExpression::new_string("foo".into()),
+                        ),
+                        (
+                            "inner3".into(),
+                            RestrictedExpression::new_record([(
+                                "innerinner".into(),
+                                RestrictedExpression::from_str(r#"Employee::"abc""#).unwrap(),
+                            )]),
+                        ),
+                    ]),
+                ),
+                (
+                    "home_ip".into(),
+                    RestrictedExpression::from_str(r#"ip("10.20.30.40")"#).unwrap(),
+                ),
+                (
+                    "work_ip".into(),
+                    RestrictedExpression::from_str(r#"ip("10.50.60.70")"#).unwrap(),
+                ),
+                (
+                    "trust_score".into(),
+                    RestrictedExpression::from_str(r#"decimal("36.53")"#).unwrap(),
+                ),
+                (
+                    "tricky".into(),
+                    RestrictedExpression::from_str(r#"{ type: "foo", id: "bar" }"#).unwrap(),
+                ),
+            ]),
+            HashSet::new(),
+        );
+        match entity.validate(&schema) {
+            Ok(_) => panic!("expected an error due to extraneous attribute"),
+            Err(e) => {
+                assert!(
+                    e.to_string().contains(r#"attribute `extra` on `Employee::"123"` should not exist according to the schema"#),
+                    "actual error message was {e}",
+                )
+            }
+        }
+
+        let entity = Entity::new(
+            EntityUid::from_strs("Manager", "jane"),
+            HashMap::new(),
+            HashSet::new(),
+        );
+        match entity.validate(&schema) {
+            Ok(_) => panic!("expected an error due to unexpected entity type"),
+            Err(e) => {
+                assert!(
+                    e.to_string().contains(r#"entity `Manager::"jane"` has type `Manager` which is not declared in the schema"#),
+                    "actual error message was {e}",
+                )
+            }
+        }
+    }
+}
+
```

Comment:
I probably missed it, but do we have test cases for the cases where?
1) action entities that exist in entities.json but not schema,
2) vice-versa, 
3) the entities are the same but not `deep_eq`